### DISCUSSION TITLE
First visit of a new page raises error? Works after refresh

### DISCUSSION THREAD

**Discussion by HiRolla** [joomlatools/joomlatools-pages#883] [Q&A] [answered]
Hi! Each time I create some new-page.html.md on the first visit I get a white page with this text:
```
Notice: Undefined index: diary/privet-dnevnik.html.md in /var/www/alex/data/www/examlpe.com/libraries/joomlatools-components/pages/page/registry.php on line 256

Notice: Trying to access array offset on value of type null in /var/www/alex/data/www/examlpe.com/libraries/joomlatools-components/pages/page/registry.php on line 256
```
repeated 6 times and
```
Fatal error: Uncaught Error: Call to a member function toArray() on int in /var/www/alex/data/www/examlpe.com/libraries/joomlatools/library/model/entity/abstract.php:84 Stack trace: #0 /var/www/alex/data/www/examlpe.com/libraries/joomlatools-components/pages/model/entity/item.php(35): KModelEntityAbstract->__construct() #1 /var/www/alex/data/www/examlpe.com/libraries/joomlatools-components/pages/model/entity/item.php(28): ComPagesModelEntityItem->__construct() #2 /var/www/alex/data/www/examlpe.com/libraries/joomlatools/library/object/manager/manager.php(721): ComPagesModelEntityItem::getInstance() #3 /var/www/alex/data/www/examlpe.com/libraries/joomlatools/library/object/manager/manager.php(161): KObjectManager->_instantiate() #4 /var/www/alex/data/www/examlpe.com/libraries/joomlatools/library/object/object.php(290): KObjectManager->getObject() #5 /var/www/alex/data/www/examlpe.com/libraries/joomlatools-components/pages/page/registry.php(257): KObject->getObject() #6 /var/www/alex/data/www/examlpe.com/libraries/joomlatool in /var/www/alex/data/www/examlpe.com/libraries/joomlatools/library/model/entity/abstract.php on line 84
```
repeated twice.

How should I figure out what might be wrong?

**Comment by johanjanssens** [comment, marked as answer]
Sorry for the delay in getting back to you. Been busy working on Joomlatools Server. 

 This is a little racing condition, that only happens during development. Pages uses an internal pages cache, it validates the cache but in some cases this doesn't reflect changes fast enough. For example I get the issue when running in Docker setup, since file changes are not synced fast enough with the host.

When doing development work, it's best to always enable debug mode in Joomla. This turns off caching optimisations. There is also a config option for this called `debug`, see: [.../code/config/config.php#L29](https://github.com/joomlatools/joomlatools-pages/blob/master/code/config/config.php#L29)

The page registry cache is turned off with debug is enabled: [.../code/page/registry.php#L45](https://github.com/joomlatools/joomlatools-pages/blob/master/code/page/registry.php#L45)

Hope that helps.
